src/static-build/loader.ts

```diff
@@ -0,0 +1,155 @@
+import getFeatures from './getFeatures';
+import { LoaderContext } from 'webpack/lib/webpack';
+const { getOptions } = require('loader-utils');
+import * as recast from 'recast';
+const types = recast.types;
+const namedTypes = types.namedTypes;
+const builders = types.builders;
+const compose = require('recast/lib/util').composeSourceMaps;
+
+/**
+ * A map of features that should be statically replaced in the code
+ */
+export interface StaticHasFeatures {
+	[feature: string]: boolean;
+}
+
+const HAS_MID = /\/has$/;
+const HAS_PRAGMA = /^\s*(!?)\s*has\s*\(["']([^'"]+)['"]\)\s*$/;
+const HAS_MID_SNIFF = /require\(["'][^,"']+\/has["']\)/;
+const HAS_PRAGMA_SNIFF = /^\s*['"]\s*(!?)\s*has\s*\(["']([^'"]+)['"]\)\s*['"]\s*;\s*$/m;
+
+export default function (this: LoaderContext, content: string, sourceMap?: { file: '' }) {
+	if (!(HAS_MID_SNIFF.test(content) || HAS_PRAGMA_SNIFF.test(content))) {
+		return content;
+	}
+	// copy features to a local scope, because `this` gets weird
+	const options = getOptions(this);
+	const { features: featuresOption, isRunningInNode } = options;
+	const args = (sourceMap && sourceMap.file && {
```

Comment:
slightly confusing variable name as `args` is often used as a parameter name for the rest of the arguments passed to the option.  Would `recastArgs` be a clearer name or `parsingArgs`?  Also it took me a bit of time to grok what was being done, because it is fairly far from the use site.